src/solids4FoamModels/numerics/triangleQuadrature/triangleQuadrature.C

```diff
@@ -0,0 +1,260 @@
+            }
+        }
+    );
+
+    // 7 point quadrature, exact for polynomials up to 5 order
+    rules_.insert
+    (
+        7,
+        quadratureRule
+        {
+            List<point>
+            {
+                point(1.0/3.0, 1.0/3.0, 1.0/3.0),
+                point(0.059715871789770, 0.470142064105115, 0.470142064105115),
+                point(0.470142064105115, 0.059715871789770, 0.470142064105115),
+                point(0.470142064105115, 0.470142064105115, 0.059715871789770),
+                point(0.797426985353087, 0.101286507323456, 0.101286507323456),
+                point(0.101286507323456, 0.797426985353087, 0.101286507323456),
+                point(0.101286507323456, 0.101286507323456, 0.797426985353087),
+            },
+            List<scalar>
+            {
+                 0.225000000000000,
+                 0.132394152788506,
+                 0.132394152788506,
+                 0.132394152788506,
+                 0.125939180544827,
+                 0.125939180544827,
+                 0.125939180544827
+            }
+        }
+    );
+
+    // 12 point quadrature, exact for polynomials up to 6 order
+    rules_.insert
+    (
+        12,
+        quadratureRule
+        {
+            List<point>
+            {
+                point(0.501426509658179, 0.249286745170910, 0.249286745170910),
+                point(0.249286745170910, 0.501426509658179, 0.249286745170910),
+                point(0.249286745170910, 0.249286745170910, 0.501426509658179),
+                point(0.873821971016996, 0.063089014491502, 0.063089014491502),
+                point(0.063089014491502, 0.873821971016996, 0.063089014491502),
+                point(0.063089014491502, 0.063089014491502, 0.873821971016996),
+                point(0.053145049844817, 0.310352451033784, 0.636502499121399),
+                point(0.053145049844817, 0.636502499121399, 0.310352451033784),
+                point(0.310352451033784, 0.053145049844817, 0.636502499121399),
+                point(0.636502499121399, 0.053145049844817, 0.310352451033784),
+                point(0.310352451033784, 0.636502499121399, 0.053145049844817),
+                point(0.636502499121399, 0.310352451033784, 0.053145049844817)
+            },
+            List<scalar>
+            {
+                0.116786275726379,
+                0.116786275726379,
+                0.116786275726379,
+                0.050844906370207,
+                0.050844906370207,
+                0.050844906370207,
+                0.082851075618374,
+                0.082851075618374,
+                0.082851075618374,
+                0.082851075618374,
+                0.082851075618374,
+                0.082851075618374
+            }
+        }
+    );
+}
+
+
+// * * * * * * * * * * * * * Private Member Functions * * * * * * * * * * * * //
+
+List<point> triangleQuadrature::barycentricToPoint(const List<point>& localGP)
+{
+    List<point> globalGP(localGP.size());
```

Comment:
Should we use a tmp here to avoid return-by-copy? This is quick to change

Reply:
Hmnn... now, I see: List is not derived from refCount so we cannot use it. Could we use Field instead of List?

Reply:
Sure, I added `tmp<Field` int the latest commit. 
I don't know why, but I used List almost everywhere xD


I forgot to pull yours commit #454209a;  #dbc0c87 is commit with tmp and #e94bdbf brings back your formatting changes. I hope I didn't screw anything up.